core/templates/dev/head/domain/topic/SubtopicObjectFactory.js

```diff
@@ -19,10 +19,15 @@
 
 oppia.factory('SubtopicObjectFactory', [
   function() {
-    var Subtopic = function(subtopicBackendObject) {
+    var Subtopic = function(subtopicBackendObject, skillIdToDescriptionMap) {
       this._id = subtopicBackendObject.id;
       this._title = subtopicBackendObject.title;
-      this._skillIds = subtopicBackendObject.skill_ids;
+      this._skills = subtopicBackendObject.skill_ids.map(function(skillId) {
+        return {
```

Comment:
We should make this an actual domain object, not just a dict. What does it represent conceptually? I guess it's not quite a skill; maybe it's a skill summary? (Not sure if there's already something else with that concept though.)

(Make sure to rename the field accordingly, e.g. this._skillSummaries -- and other methods/functions that rely on this object. I think we should reserve the name Skill for the full thing.)

Reply:
Done, created a SkillSummary object.